LibreHardwareMonitorLib/Hardware/Motherboard/Motherboard.cs

```diff
@@ -71,11 +72,29 @@ public Motherboard(SMBios smBios, ISettings settings)
             superIO = _lpcIO.SuperIO;
         }
 
+
         EmbeddedController embeddedController = EmbeddedController.Create(Model, settings);
 
         SubHardware = new IHardware[superIO.Count + (embeddedController != null ? 1 : 0)];
+
+        // there may be more than 1 of the same SuperIO chip
+        Dictionary<Chip, int?> currentIndexes = _lpcIO.SuperIO
+            .GroupBy(x => x.Chip)
+            .ToDictionary(
+                x => x.Key,
+                x => x.Count() > 1 ? 0 : default(int?));
+
         for (int i = 0; i < superIO.Count; i++)
-            SubHardware[i] = new SuperIOHardware(this, superIO[i], Manufacturer, Model, settings);
+        {
+            ISuperIO currentSuperIO = superIO[i];
+            int? index = currentIndexes[currentSuperIO.Chip];
+
+            // there is more than 1 of this superIO chip, increment the counter
+            if (index.HasValue)
+                currentIndexes[currentSuperIO.Chip] = index.Value + 1;
+
+            SubHardware[i] = new SuperIOHardware(this, currentSuperIO, Manufacturer, Model, settings, index);
```

Comment:
Can't **index** just be assigned to **i**? Seems like an easier solution.

Reply:
> Decided not to modify identifiers if there is only 1 of that SuperIO chip to not break compatibility.

That's the tricky part. I use a nullable<int>, where null means there is only a single of that chip, and non-null which starts at 0 means there are multiple.   If there is only a single chip of a kind, I don't put the index in the identifier.  That way only  setups with duplicates of a chip will get indexes in the identifier.   Of course I'm biased towards FanControl for that specific use-case , otherwise it would break all serialized configs 😅

Reply:
I may find a way around the problem in FanControl for the identifiers to convert them back to non-indexed if the index == 0 for example. 

Gave you the reason, your call.

Reply:
> I may find a way around the problem in FanControl for the identifiers to convert them back to non-indexed if the index == 0 for example.

That sounds like a better solution and can keep this code simple.

Reply:
Commited a simplified version, looks much better.